django/db/models/fields/files.py

```diff
@@ -312,6 +314,15 @@ def get_prep_value(self, value):
 
     def pre_save(self, model_instance, add):
         file = super().pre_save(model_instance, add)
+        if file.name is None and file._file is not None:
```

Comment:
```suggestion
if not file.name and file._file is not None:
```
How about this?
Because passing, for example, an empty string also creates a database record with an empty file path, which leads to the problems already described.

Reply:
if it is possible for you to write a test that causes the problem you described, it think it is better to create a new ticket so that you could provide a patch. If you are new to Django contribution, please refer to https://code.djangoproject.com/.